library/src/main/java/com/novoda/downloadmanager/lib/BatchQuery.java

```diff
@@ -125,6 +126,17 @@ public Builder withStatusFilter(@Status int statusFlags) {
             return this;
         }
 
+        /**
+         * Filter by extra data.
+         *
+         * @return {@link BatchQuery.Builder}
+         */
+        public Builder withExtraData(String extraData) {
```

Comment:
No test for this :panda_face: https://github.com/novoda/download-manager/blob/master/library/src/test/java/com/novoda/downloadmanager/lib/BatchQueryTest.java